From 6e14c1b46829ba881df7aa5a0c7ff8a2fc69f5c7 Mon Sep 17 00:00:00 2001
From: Juan Manuel Lopez Carrillo <juanmanuellopezcarrillo@gmail.com>
Date: Wed, 8 Jul 2026 01:39:34 +0200
Subject: [PATCH 008/130] dts: opi4a: use the BSP RGMII delays (rgmii +
 tx100/rx300ps)

The mainline values (rgmii-id, tx0/rx300) gave a 1Gbps link but 0 traffic
(ARP ok, ping ko) = double delay (internal PHY + syscon). The BSP uses
phy-mode rgmii (no PHY delay) with tx-delay=1/rx-delay=3 in the MAC syscon.
Mapped: tx-internal-delay-ps=100, rx-internal-delay-ps=300.
---
 arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
index 369529183..0f3e715c2 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
@@ -493,10 +493,13 @@ &usbphy {
 };
 
 &gmac1 {
-	phy-mode = "rgmii-id";
+	/* BSP delays (verified on this board): phy-mode rgmii, no
+	 * internal PHY delay; the MAC syscon provides the skew
+	 * (BSP tx-delay=1 -> 100ps, rx-delay=3 -> 300ps). */
+	phy-mode = "rgmii";
 	phy-handle = <&ext_rgmii_phy>;
 	phy-supply = <&reg_cldo4>;
-	tx-internal-delay-ps = <0>;
+	tx-internal-delay-ps = <100>;
 	rx-internal-delay-ps = <300>;
 	status = "okay";
 };
